src/stateless/slider.vue

```diff
@@ -0,0 +1,597 @@
+
+        if (this.value < this.min || this.max < this.value)
+            throw new Error('Value must be between min and max')
+    },
+    mounted () {
+        this.bounds = this.$refs.slider.getBoundingClientRect()
+        this.handle = this.$refs.handle.getBoundingClientRect()
+
+        // threshold for ticks disappearing under labels,  +/- 5 guarantees some padding
+        this.minThreshold = this.$refs.min.getBoundingClientRect().width + 5
+        this.maxThreshold = this.bounds.width - this.$refs.max.getBoundingClientRect().width - 5
+
+        this.domReady = true
+    },
+    computed: {
+        limitValue () {
+            return this.limit || this.max
+        },
+        minLabelValue () {
+            return this.minLabel || this.min.toString()
+        },
+        maxLabelValue () {
+            return this.maxLabel || this.limitValue.toString()
+        },
+        stepsCount () {
+            return (this.limitValue - this.min) / this.step
+        },
+        stepPercentage () {
+            return 1 / this.stepsCount
+        },
+        percentage () {
+            return this.index * this.stepPercentage
+        },
+        position () {
+            if (this.domReady) 
+                return Math.min(this.index, this.stepsCount) * this.stepPercentage * this.bounds.width
+        },
+        activeBarStyle () {
+            return {
+                'width': `${this.position}px`
+            }
+        },
+        handleStyle () {
+            return {
+                'left': `${this.position}px`
+            }
+        },
+        decimalPlacesCount () {
+            let decimals = this.step.toString().split('.')[1]
+            return decimals ? decimals.length : 0
+        },
+        currentValue () {
+          return this.inputFlag ? this.inputValue : Number((this.min + this.index * this.step).toFixed(this.decimalPlacesCount))
+        },
+        inputType () {
+            return this.decimalPlacesCount === 0 ? 'number' : 'float'
+        }
+    },
+    methods: {
+        startDrag (e) {
+            if (this.disabled) return
+
+            this.bounds = this.$refs.slider.getBoundingClientRect() // hotfix, will try to find a better way
+            this.handle = this.$refs.handle.getBoundingClientRect() // hotfix, will try to find a better way
+
+            this.changedFlag = true
+            this.draggingFlag = true
+            this.inputFlag = false
+
+            this.setPosition(e)
+            window.addEventListener('mouseup', this.stopDrag)
+            window.addEventListener('mousemove', this.setPosition)
+            this.$refs.slider.focus()
+        },
+        setPosition (e) {
+            let percent = (e.clientX - this.bounds.x) / this.bounds.width
+            percent = Math.min(Math.max(0, percent), 0.999999)
+
+            let edgeStepOffset = this.stepPercentage / 2
+            this.index = Math.floor((percent - edgeStepOffset) / this.stepPercentage) + 1
```

Comment:
Instead of changing the index (which should probably be a computed property) you can just emit `input` event and set the new value this way.